### PR TITLE
Milestone2

### PR DESCRIPTION
**Pull request by danielRM88**
Lots of new features!

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,6 +27,13 @@ gem 'jbuilder', '~> 2.0'
 # bundle exec rake doc:rails generates the API under doc/api.
 gem 'sdoc', '~> 0.4.0', group: :doc
 
+gem 'will_paginate', '~> 3.0.6'
```
Comment (mmorava) [inline Gemfile:30]: will_paginate is not bad, actually its written by my friends, however i recomend you to have a look at the Kaminari gem as well

File: app/assets/javascripts/categories.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (mmorava) [inline app/assets/javascripts/categories.coffee:1]: We should remove JS files that scaffold generator generates for us if we do not put any js code in them. they are just poluting the worktree

File: app/models/product.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+#  updated_at  :datetime         not null
+#  category_id :integer
+#  description :text
+#  url         :string
+#
+
+class Product < ActiveRecord::Base
+  belongs_to :category
+
+  scope :with_category_id, -> (category_id) { where(category_id: category_id) }
```
Comment (mmorava) [inline app/models/product.rb:18]: what if we put category_id param to nil? Should we then return all the products, or only the products that have a nil as a category?  should we allow product category to be nil? Nothing major just something to think about

File: app/views/categories/_form.html.slim [hunk 4]
```diff
@@ -0,0 +1,19 @@
+= form_for(@category) do |f|
```
Comment (mmorava) [inline app/views/categories/_form.html.slim:1]: take a look at simple_form_for gem to simplify your life

File: config/initializers/simple_form.rb [hunk 5]
```diff
@@ -0,0 +1,166 @@
+# Use this setup block to configure all options available in SimpleForm.
+SimpleForm.setup do |config|
```
Comment (mmorava) [inline config/initializers/simple_form.rb:2]: now im confused..you introduced simple form, but you are still keep using form_for?

**Comment by mmorava** [comment]
Look good to me...i would only like it to be consistent are we using simple_form_for or form_for and then use the same helper through the whole project.

Anyway very good work, :+1:

File: config/initializers/simple_form.rb (hunk 5 above)
Comment (danielRM88) [inline config/initializers/simple_form.rb:2]: this is a generated file from rails command line... I forgot to change it

**Pull request merged**